zebra-state/src/service/finalized_state/disk_format/upgrade.rs

```diff
@@ -353,10 +366,17 @@ impl DbFormatChange {
 
             // Set up task for reading orchard note commitment trees
             let db = upgrade_db.clone();
+            let should_cancel_flag = should_cancel_format_change;
             let orchard_read_task = std::thread::spawn(move || {
                 for (height, tree) in
                     db.orchard_tree_by_height_range(orchard_height..initial_tip_height)
                 {
+                    // Breaking from this loop and dropping the orchard_tree channel
+                    // will cause the orchard compare and delete tasks to finish.
+                    if should_cancel_flag.load(atomic::Ordering::Relaxed) {
```

Comment:
Using `Relaxed` ordering here because it doesn't matter in this case if the cancel signal is seen out of order.